tests/Doctrine/Tests/DBAL/Platforms/SqlitePlatformTest.php

```diff
@@ -807,4 +808,28 @@ public function testGetCreateTableSQLWithColumnCollation(): void
             $this->platform->getCreateTableSQL($table)
         );
     }
+
+    public function testRemovingOrphanedForeignKeys(): void
```

Comment:
Why is this tested only for SQLite? Is it not relevant to other platforms? When it comes to fixing platform-specific issues, I'd much rather prefer having a functional test first. A unit test is only needed for faster feedback.

Reply:
Only SQLite platform has the `supportsCreateDropForeignKeyConstraints` returning false. I could move it to AbstractPlatform test and check for the supports flag to future proof this.